### PR TITLE
Replace ConfigureLogging(Action<ILoggerFactory>) with ConfigureLogging(Action<LoggerFactory>)

### PR DESCRIPTION
**Pull request by JunTaoLuo**
addresses #1007

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs [hunk 1]
```diff
@@ -82,10 +88,16 @@ public interface IWebHostBuilder
...
+        /// This method is obsolete and will be removed in a future version. The recommended alternative is <see cref="UseLoggerFactory(Func{WebHostBuilderContext, ILoggerFactory})"/>.
+        /// </para>
+        /// <para>
         /// Specify the <see cref="ILoggerFactory"/> to be used by the web host.
+        /// </para>
         /// </summary>
         /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to be used.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.AspNetCore.Hosting.IwebHostBuilder.UseLoggerFactory(Func<WebHostBuilderContext, ILoggerFactory> createLoggerFactory).")]
         IWebHostBuilder UseLoggerFactory(ILoggerFactory loggerFactory);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs]: I'm not sure we need to deprecate this one.

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 2]
```diff
@@ -90,10 +90,17 @@ public IWebHostBuilder UseSetting(string key, string value)
...
+        /// This method is obsolete and will be removed in a future version. The recommended alternative is <see cref="UseLoggerFactory(Func{WebHostBuilderContext, ILoggerFactory})"/>.
+        /// </para>
+        /// <para>
         /// Specify the <see cref="ILoggerFactory"/> to be used by the web host.
+        /// </para>
         /// </summary>
         /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to be used.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.AspNetCore.Hosting.IwebHostBuilder.UseLoggerFactory(Func<WebHostBuilderContext, ILoggerFactory> createLoggerFactory).")]
+
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: nit:spacing

File: src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs (hunk 1 above)
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs]: I believe you are right. @glennc when I originally filed the issue I remember we said we wanted to deprecate both but I don't see why we want to force anyone to use the overload with the context if it's not needed.

**Comment by JunTaoLuo** [comment]
@davidfowl the issue has be triaged to preview2. Do you think that is okay or should we pull it back to preview1?

**Comment by JunTaoLuo** [comment]
ping @muratg @davidfowl @BrennanConroy @Tratcher since we want to pull this change back to preview1.

File: src/Microsoft.AspNetCore.Hosting.Abstractions/Microsoft.AspNetCore.Hosting.Abstractions.csproj [hunk 3]
```diff
@@ -20,6 +20,7 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="$(AspNetCoreVersion)" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="$(AspNetCoreVersion)" />
     <PackageReference Include="Microsoft.Extensions.FileProviders.Abstractions" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="$(AspNetCoreVersion)" />
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting.Abstractions/Microsoft.AspNetCore.Hosting.Abstractions.csproj]: hmmm

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Hosting.Abstractions/Microsoft.AspNetCore.Hosting.Abstractions.csproj]: Oops this needs to be moved to extensions.

Comment (cwe1ss) [inline src/Microsoft.AspNetCore.Hosting.Abstractions/Microsoft.AspNetCore.Hosting.Abstractions.csproj]: That's a no-go isn't it?!

**Pull request merged**